economy/models.py

```diff
@@ -48,6 +48,13 @@ def transaction_history(self) -> Dict[str, Union[QuerySet, 'ProductOrder', 'Tran
             'deposits': self.deposits.all()
         }
 
+    @property
+    def money_spent(self) -> int:
+        purchase_totals = sum(order.product.price for order in self.transaction_history["product_orders"].all())
+        transfers = sum(transfer.amount for transfer in self.source_transfers.all())
+        return purchase_totals + transfers
+
+
```

Comment:
Should probably look into django db aggregations seeing as this will probably become extremely slow as the dataset grows, see https://docs.djangoproject.com/en/3.1/topics/db/aggregation/#generating-aggregates-over-a-queryset